conn.go

```diff
@@ -179,66 +185,113 @@ func Connect(addr string, cfg ConnConfig, pool ConnectionPool) (*Conn, error) {
 	}
 
 	for i := 0; i < cfg.NumStreams; i++ {
+		c.calls[i].resp = make(chan error, 1)
 		c.uniq <- i
 	}
 
+	go c.serve()
+
 	if err := c.startup(&cfg); err != nil {
 		conn.Close()
 		return nil, err
 	}
-
-	go c.serve()
+	c.started = true
```

Comment:
Just as a point of interest, do we know that the connection has actually been started at this point in time, since `c.serve()` is executed asynchronously? Or is this irrelevant?